### PR TITLE
Added CHECKPOINT_SYNC_URL environment variable

### PR DESCRIPTION
**Pull request by Ajaypathak372**
Fixes | Closes | Resolves #

> Anything marked as optional that you didn't need to fill in, please remove it from the PR description. Choose one of the keywords above to refer to the issue this PR solves, followed by the issue number (e.g Fixes # 666). If there is no issue, remove the line. Remove this note after reading.
## Changes:
-
## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [ ] Yes
- [x] No

**In case you checked yes, did you write tests?**

- [ ] Yes
- [ ] No

**Comments about testing , should you have some** (optional)

## Further comments (optional)

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by AntiD2ta — COMMENTED** [review]
Your approach works, but it may not be the best because we have the checkpoint sync flag duplicated in the template. I can't think right now on a better approach though. This is a new case, where we would want to use the .env variable but if we get some input then use the input instead. I will approve this PR, but after we close the PR that removes the mainnet templates, because you need those changes.

**Comment by Ajaypathak372** [comment]
I have updated the code, and now it becomes if-else  and not duplicating `checkpoint-sync-url`. As of now, I have make the changes only for the `lodestar` merge consensus, please review the latest commit.

File: templates/services/merge/consensus/lighthouse.tmpl [hunk 1]
```diff
@@ -49,9 +49,8 @@
       - --import-all-attestations
       - --metrics
       - --metrics-port={{.ClMetricsPort}}
-      - --metrics-address=0.0.0.0{{if .ClCheckpointSyncUrl}}
-      - --checkpoint-sync-url=${CHECKPOINT_SYNC_URL}{{end}}{{with .CheckpointSyncUrl}}
-      - --checkpoint-sync-url={{.}}{{end}}{{range $flag := .ClExtraFlags}}
+      - --metrics-address=0.0.0.0{{if or .ClCheckpointSyncUrl .CheckpointSyncUrl}}
+      - --checkpoint-sync-url={{if .CheckpointSyncUrl}}{{ .CheckpointSyncUrl }}{{else if .ClCheckpointSyncUrl}}${CHECKPOINT_SYNC_URL}{{end}}{{end}}{{range $flag := .ClExtraFlags}}
```
Comment (cbermudez97) [inline templates/services/merge/consensus/lighthouse.tmpl:53]: Using `{{else}}` should be enough here. It will work tough.

**Review by cbermudez97 — APPROVED** [review]

File: templates/services/merge/consensus/lighthouse.tmpl (hunk 1 above)
Comment (Ajaypathak372) [reply templates/services/merge/consensus/lighthouse.tmpl]: Done

**Pull request merged**